app/jobs/rate_limit_checker_job.rb

```diff
@@ -1,8 +1,8 @@
 class RateLimitCheckerJob < ApplicationJob
   queue_as :rate_limit_checker
 
-  def perform(user_id)
+  def perform(user_id, action)
```

Comment:
Do you have a reason for different argument names: `action` vs `situation`? I find it a bit confusing.

Reply:
I started using `action` but came across the switch statement using `situation`, so I went along with that. Yeah this is confusing. 

I've went a bit further and changed `limit_by_situation` to `limit_by_action`, since that makes more sense.